### PR TITLE
Added charts

### PR DESCRIPTION
**Pull request by Kiran2821996**
Created Charts

### CODE REVIEW

**Comment by Kiran2821996** [comment]
review

File: src/App.js [hunk 1]
```diff
@@ -2,11 +2,15 @@
 import './App.css';
```
Comment (Souvikdeb2612) [inline src/App.js]: css imports should be in the end and keep a space above that

File: src/components/Trending.jsx [hunk 2]
```diff
@@ -1,78 +1,106 @@
...
        
-        dataKey="item.price_btc"
-        stroke="#8884d8"
-        fill="#8884d8"
-        fillOpacity={0.6}
-      />
-    </RadarChart>
-        </div>
+        
+        <div style={{ display: "flex",height:"20vh",flexWrap:"wrap",width:"70%",alignItems:"center",justifyContent:"center"}}>
```
Comment (Souvikdeb2612) [inline src/components/Trending.jsx]: remove inline styling and only use external stylesheet

File: src/components/Trending.jsx [hunk 3]
```diff
@@ -1,78 +1,106 @@
...
+     
+      <div
+        style={{
+          display: "flex",
+          justifyContent: "center",
+          alignItems: "center",
+          width: "100vw",
+        }}
+      >
+        <AreaChart
```
Comment (Souvikdeb2612) [inline src/components/Trending.jsx]: Create this as a separate component under charts folder and pass data as props

File: src/components/Trending.jsx [hunk 4]
```diff
@@ -1,78 +1,106 @@
...
-            )
+            
+              </div>
+            
+          );
         })}
+      </div>
+        
+     
+      <div
```
Comment (Souvikdeb2612) [inline src/components/Trending.jsx]: remove inline styling everywhere

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 1 above)
Comment (Kiran2821996) [reply src/App.js]: okay sir

**Comment by Kiran2821996** [comment]
routes added

File: src/components/Trending.jsx (hunk 3 above)
Comment (Kiran2821996) [reply src/components/Trending.jsx]: okay sir by tmwr

File: src/components/Trending.jsx (hunk 4 above)
Comment (Kiran2821996) [reply src/components/Trending.jsx]: done sir

File: src/App.js [hunk 5]
```diff
@@ -1,24 +1,35 @@
-import logo from './logo.svg';
-import './App.css';
+import CoinList from "./components/coinList/CoinList";
+import News from "./components/news/News";
+import Trending from "./components/trending/Trending";
+import "./App.css";
```
Comment (Souvikdeb2612) [inline src/App.js]: First all external imports, then 1 space, all internal file and component imports, 1 space and finally css import should occur

File: src/App.js [hunk 6]
```diff
@@ -1,24 +1,35 @@
...
+      </Header>
+      <Layout>
+        <Sider>
+          <News />
+        </Sider>
+        <Content>
+          <HeaderNav />
+        <Routes>
+        <Route path="/" element={<Trending />} />
+        <Route path="/shivansh-component" />
```
Comment (Souvikdeb2612) [inline src/App.js]: path name isn't meaningful

File: src/App.js [hunk 7]
```diff
@@ -1,24 +1,35 @@
...
+        <Route path="/ketan-component" />
```
Comment (Souvikdeb2612) [inline src/App.js]: same as above

File: src/components/coinList/CoinList.jsx [hunk 8]
```diff
@@ -0,0 +1,76 @@
+import React,{useState,useEffect} from 'react'
+// import { BarChart, Bar } from "recharts";
+import axios from 'axios';
+import './Coinlist.css'
```
Comment (Souvikdeb2612) [inline src/components/coinList/CoinList.jsx]: Format imports like above

File: src/components/coinList/CoinList.jsx [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+function CoinList() {
+    const [data, setData] = useState([]);
+
+  useEffect(() => {
+    axios
+      .get("https://api.coingecko.com/api/v3/coins/markets?vs_currency=inr&order=market_cap_desc&per_page=100&page=1&sparkline=false")
+      .then((response) => {
+        setData([...response.data]);
+        console.log(response.data)
```
Comment (Souvikdeb2612) [inline src/components/coinList/CoinList.jsx]: remove console statement before raising pr

File: src/components/coinList/CoinList.jsx [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+      });
+  }, []);
+  return (
+    <div >
+       <Carousel autoplay="true" dots="false">
+       {data.map((item) => {
+      return (
+        
+          <div style={contentStyle}>
+            <div  style={{display:"flex",justifyContent: "center",
```
Comment (Souvikdeb2612) [inline src/components/coinList/CoinList.jsx]: remove inline styling

File: src/components/coinList/CoinList.jsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+          alignItems: "center",marginTop:"1%"}}>
+            <img src={item.image} width={50} alt="" />
```
Comment (Souvikdeb2612) [inline src/components/coinList/CoinList.jsx]: add alt text

File: src/components/header/HeaderNav.jsx [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import {  NavLink } from "react-router-dom";
+import "./HeaderNav.css";
```
Comment (Souvikdeb2612) [inline src/components/header/HeaderNav.jsx:4]: space before css import

File: src/components/news/News.jsx [hunk 13]
```diff
@@ -0,0 +1,49 @@
+import React,{useEffect,useState} from 'react';
+import axios from 'axios';
+import './News.css';
+import { Carousel } from 'antd';
```
Comment (Souvikdeb2612) [inline src/components/news/News.jsx]: format imports

File: src/components/news/News.jsx [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+function News() {
+  const [data, setData] = useState([]);
+    
+  useEffect(() => {
+    axios
+      .get(
+        "https://newsapi.org/v2/everything?q=crypto&from=2022-10-11&sortBy=publishedAt&apiKey=72635734611e40d5b63c90dc5248f92b"
+      )
+      .then((response) => {
+        console.log(response.data);
```
Comment (Souvikdeb2612) [inline src/components/news/News.jsx]: remove console statement

File: src/components/trending/Trending.jsx [hunk 15]
```diff
@@ -0,0 +1,94 @@
...
+  AreaChart,
+  Area,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Tooltip,
+  ResponsiveContainer,
+} from "recharts";
+import axios from "axios";
+import "./Trending.css";
```
Comment (Souvikdeb2612) [inline src/components/trending/Trending.jsx:19]: space before css import

File: src/components/trending/Trending.jsx [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+            stroke="#457B9D"
+            fill="#A8DADC"
+            fillOpacity={0.6}
+          />
+        </RadarChart>
+      </div>
+      <div className="trend_block">
+          {data.map((item) => {
+            return (
+              <div className="trend_block_item" style={{ display: "flex" }}>
```
Comment (Souvikdeb2612) [inline src/components/trending/Trending.jsx]: create a separate component and import it here

File: src/index.js [hunk 17]
```diff
@@ -3,11 +3,16 @@ import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import 'antd/dist/antd.css';
```
Comment (Souvikdeb2612) [inline src/index.js:6]: format imports

File: src/App.js [hunk 18]
```diff
@@ -11,8 +11,9 @@ const { Header, Sider, Content } = Layout;
 
 function App() {
   return (
-    <Layout>
-      <Header>
+    <div  >
+ <Layout className="main">
+      <Header style={{height:"12vh",paddingBottom:"1%"}}>
```
Comment (Souvikdeb2612) [inline src/App.js]: remove inline styling

**Review by Souvikdeb2612 — CHANGES_REQUESTED** [review]

File: src/App.js (hunk 6 above)
Comment (Kiran2821996) [reply src/App.js]: edited

File: src/App.js (hunk 7 above)
Comment (Kiran2821996) [reply src/App.js]: edited

File: src/components/coinList/CoinList.jsx (hunk 8 above)
Comment (Kiran2821996) [reply src/components/coinList/CoinList.jsx]: doe sir

File: src/components/coinList/CoinList.jsx (hunk 9 above)
Comment (Kiran2821996) [reply src/components/coinList/CoinList.jsx]: removed

File: src/components/coinList/CoinList.jsx (hunk 10 above)
Comment (Kiran2821996) [reply src/components/coinList/CoinList.jsx]: done sir

File: src/components/header/HeaderNav.jsx (hunk 12 above)
Comment (Kiran2821996) [reply src/components/header/HeaderNav.jsx:4]: done sir

File: src/components/news/News.jsx (hunk 13 above)
Comment (Kiran2821996) [reply src/components/news/News.jsx]: done sir

File: src/components/news/News.jsx (hunk 14 above)
Comment (Kiran2821996) [reply src/components/news/News.jsx]: done sir

File: src/components/trending/Trending.jsx (hunk 15 above)
Comment (Kiran2821996) [reply src/components/trending/Trending.jsx:19]: done sir

File: src/index.js (hunk 17 above)
Comment (Kiran2821996) [reply src/index.js:6]: done

File: src/App.js (hunk 18 above)
Comment (Kiran2821996) [reply src/App.js]: its bit tricky!!!

**Review by Souvikdeb2612 — APPROVED** [review]

**Pull request merged**